Compiling a custom build fails.

I am trying to make a custom ESP32_4M316k build.  (Also tried for S3)
To rule out any mistake by me, I used the custom sample, but I am getting the following error message (it’s a sample of the first few lines):

```
Compiling .pio/build/custom_ESP32_4M316k/src/ESPEasy/net/DataTypes/EthernetParameters_ESP32P4.cpp.o
In file included from src/src/ESPEasyCore/../Globals/CPlugins.h:7,
                 from src/src/ESPEasyCore/Controller.h:7,
                 from src/_Plugin_Helper.h:15,
                 from /Users/X/Documents/GitHub/ESPEasy/src/ESPEasy.ino:100:
src/src/ESPEasyCore/../Globals/../DataStructs/ControllerSettingsStruct.h:197:3: error: 'TLS_types' does not name a type
  197 |   TLS_types TLStype() const { return static_cast<TLS_types>(VariousBits1.TLStype); }
      |   ^~~~~~~~~
src/src/ESPEasyCore/../Globals/../DataStructs/ControllerSettingsStruct.h:198:21: error: 'TLS_types' has not been declared
  198 |   void      TLStype(TLS_types tls_type) { VariousBits1.TLStype = static_cast<uint8_t>(tls_type); }
      |                     ^~~~~~~~~
src/src/ESPEasyCore/../Globals/../DataStructs/ControllerSettingsStruct.h:201:36: error: 'TLS_types' has not been declared
  201 |   String    getCertificateFilename(TLS_types tls_type) const;
```

And while we are still discussing failed custom build, here's the latest error with the _custom_ESP32c3_4M316k_ env:

```
Compiling .pio\build\custom_ESP32c3_4M316k\src\src\WebServer\SettingsArchive.cpp.o
src/src/WebServer/Markup.cpp: In function 'void addPinSelect(PinSelectPurpose, const String&, int)':
src/src/WebServer/Markup.cpp:937:13: error: 'isPSRAMInterfacePin' was not declared in this scope
  937 |             isPSRAMInterfacePin(gpio) ? getConflictingUse_wrapped(gpio, purpose, true) : F("")
      |             ^~~~~~~~~~~~~~~~~~~
Compiling .pio\build\custom_ESP32c3_4M316k\FrameworkArduino\USBMSC.cpp.o
src/src/WebServer/Markup_Forms.cpp: In function 'void addFormPinStateSelect(int, int)':
src/src/WebServer/Markup_Forms.cpp:622:9: error: 'isPSRAMInterfacePin' was not declared in this scope
  622 |     if (isPSRAMInterfacePin(gpio)) {
      |         ^~~~~~~~~~~~~~~~~~~
*** [.pio\build\custom_ESP32c3_4M316k\src\src\WebServer\Markup.cpp.o] Error 1
*** [.pio\build\custom_ESP32c3_4M316k\src\src\WebServer\Markup_Forms.cpp.o] Error 1
```

--- a/src/src/WebServer/Markup.cpp
+++ b/src/src/WebServer/Markup.cpp
@@ -12,6 +12,7 @@
 #include "../Helpers/Convert.h"
 #include "../Helpers/ESPEasy_UnitOfMeasure.h"
 #include "../Helpers/Hardware_GPIO.h"
+#include "../Helpers/Hardware_device_info.h"
 #include "../Helpers/StringConverter_Numerical.h"
 #include "../Helpers/StringConverter.h"
 #include "../Helpers/StringGenerator_GPIO.h"

--- a/src/src/WebServer/Markup_Forms.cpp
+++ b/src/src/WebServer/Markup_Forms.cpp
@@ -8,6 +8,7 @@
 #include "../Globals/Settings.h"
 
 #include "../Helpers/Hardware_GPIO.h"
+#include "../Helpers/Hardware_device_info.h"
 #include "../Helpers/Numerical.h"
 #include "../Helpers/StringConverter.h"
 #include "../Helpers/StringGenerator_GPIO.h"

